File: ruby/redis-rb/sentinelTLS.rb
require 'redis'

if !(ARGV.length == 3 || ARGV.length == 4)
    puts "Usage: host port service password"
    return
end

SENTINEL_HOST = ARGV[0]
PORT = ARGV[1]
SERVICE = ARGV[2]
PASSWORD = ARGV.length == 4 ? ARGV[3]:nil
#you can add more sentinels here
SENTINELS = [{host:SENTINEL_HOST, port:PORT}]
begin 
    redis = Redis.new(host: SERVICE, sentinels: SENTINELS, role: :master, password:PASSWORD,ssl:true,:ssl_params => {
        :ca_file => "../../testscripts/tls/ca_cert.pem",
        :cert    => OpenSSL::X509::Certificate.new(File.read("../../testscripts/tls/db_cert.pem")),
        :key     => OpenSSL::PKey::RSA.new(File.read("../../testscripts/tls/db_key.pem")),
        :verify_mode => OpenSSL::SSL::VERIFY_NONE
      })
    print "Set: ", redis.set("foo", "bar")
    puts
    print "Get:" ,redis.get("foo")
    puts
    #close connection at end of application
    redis.close
rescue StandardError => e
    puts e.inspect
end

File: node.js/ioredis/sentineltls.js
'use strict';
if (!(process.argv.length === 5 || process.argv.length === 6)) {
    console.log("Usage: sentinelhost sentinelport service password")
    process.exit(1)
}

const Redis = require("ioredis");
const fs = require('fs');
const tls = require('tls');


var sentinelHost = process.argv[2];
var sentinelPort = process.argv[3];
var service = process.argv[4];
var password = undefined;
if (process.argv.length === 6) {
    password = process.argv[5];
}
var tlsOpts = {
    key: fs.readFileSync('../../testscripts/tls/db_key.pem'),
    cert: fs.readFileSync('../../testscripts/tls/db_cert.pem'),
    ca: [fs.readFileSync('../../testscripts/tls/ca_cert.pem')],
    //enableTrace: true,
    checkServerIdentity: (hostname, cert) => { return null; }, //return null if you do not want to verify hostname
};

var redis = new Redis({
    sentinels: [{
        host: sentinelHost,
        port: sentinelPort
    }], //you can add more sentinels here
    enableTLSForSentinelMode: true,
    sentinelTLS: {
        ca: [fs.readFileSync('../../testscripts/tls/ca_cert.pem')],
        checkServerIdentity: (hostname, cert) => { return null; }
    },
    name: service,
    password: password,
    tls: tlsOpts,
});

redis.on("error", function (error) {
    console.error("Error in connection");
    console.error(error);
});

redis.set("foo", "bar", function (err, result) {
    if (err) {
        return console.error(err);
    }
    console.log("SET:", result);
});


redis.get("foo").then(function (result) {
    console.log("GET:", result);
});
//close connection
setTimeout(function () { redis.quit() }, 3000);